test/e2e/spidercoordinator/spidercoordinator_test.go

```diff
@@ -0,0 +1,436 @@
+				spc, err := GetSpiderCoordinator(common.SpidercoodinatorDefaultName)
+				Expect(err).NotTo(HaveOccurred(), "failed to get SpiderCoordinator, error is %v", err)
+
+				if spc.Status.Phase != coordinatormanager.Synced {
+					GinkgoWriter.Printf("status.Phase status is still synchronizing, status %+v \n", spc.Status.Phase)
+					return false
+				}
+
+				if spc.Status.OverlayPodCIDR != nil {
+					GinkgoWriter.Printf("status.overlayPodCIDR status is still synchronizing, status %+v \n", spc.Status.OverlayPodCIDR)
+					return false
+				}
+
+				GinkgoWriter.Println("status.overlayPodCIDR is nil, as expected.")
+				return true
+			}, common.ExecCommandTimeout, common.ForcedWaitingTime).Should(BeTrue())
+		})
+	})
+
+	Context("It can get the clusterCIDR from kubeadmConfig and kube-controller-manager pod", Label("V00009"), func() {
+
+		var spc *spiderpoolv2beta1.SpiderCoordinator
+		var cm *corev1.ConfigMap
+		var err error
+		if !common.CheckRunOverlayCNI() {
+			GinkgoWriter.Println("This environment is in underlay mode.")
+			Skip("Not applicable to underlay mode")
+		}
+
+		if !common.CheckCalicoFeatureOn() {
+			GinkgoWriter.Println("The CNI isn't calico.")
+			Skip("This case only run in calico")
+		}
+
+		cm, err = frame.GetConfigmap("kubeadm-config", "kube-system")
+		Expect(err).NotTo(HaveOccurred())
+
+		spc, err = GetSpiderCoordinator(common.SpidercoodinatorDefaultName)
+		Expect(err).NotTo(HaveOccurred(), "failed to get SpiderCoordinator, error is %v", err)
+
+		// Switch podCIDRType to `cluster`.
+		spcCopy := spc.DeepCopy()
+		spcCopy.Spec.PodCIDRType = pointer.String(common.PodCIDRTypeCluster)
+		Expect(PatchSpiderCoordinator(spcCopy, spc)).NotTo(HaveOccurred())
+
+		defer func() {
+			spc, err := GetSpiderCoordinator(common.SpidercoodinatorDefaultName)
+			Expect(err).NotTo(HaveOccurred(), "failed to get SpiderCoordinator, error is %v", err)
+			GinkgoWriter.Printf("Display the default spider coordinator information: %+v \n", spc)
+
+			// Switch podCIDRType to `auto`.
+			spcCopy := spc.DeepCopy()
+			spcCopy.Spec.PodCIDRType = pointer.String(common.PodCIDRTypeAuto)
+			Expect(PatchSpiderCoordinator(spcCopy, spc)).NotTo(HaveOccurred())
+
+			Eventually(func() bool {
+
+				if spc.Status.OverlayPodCIDR == nil || spc.Status.Phase != coordinatormanager.Synced {
+					GinkgoWriter.Printf("status.overlayPodCIDR status is still synchronizing, status %+v \n", spc.Status.OverlayPodCIDR)
+					return false
+				}
+				for _, cidr := range spc.Status.OverlayPodCIDR {
+					if ip.IsIPv4CIDR(cidr) {
+						Expect(cidr).To(Equal(v4PodCIDRString))
+						GinkgoWriter.Printf("ipv4 podCIDR is as expected, value %v=%v \n", cidr, v4PodCIDRString)
+					} else {
+						Expect(cidr).To(Equal(v6PodCIDRString))
+						GinkgoWriter.Printf("ipv6 podCIDR is as expected, value %v=%v \n", cidr, v6PodCIDRString)
+					}
+				}
+				return true
+			}, common.ExecCommandTimeout, common.ForcedWaitingTime).Should(BeTrue())
+		}()
```

Comment:
377 行此处格式有误。 `}()` 错误的用法，正确的应该是 `})`, 并且它还缺少 })。格式应该如下？

```
	Context("It can get the clusterCIDR from kubeadmConfig and kube-controller-manager pod", Label("V00009"), func() {

		var spc *spiderpoolv2beta1.SpiderCoordinator
		var cm *corev1.ConfigMap
		var err error
		BeforeEach(func() {
			if !common.CheckRunOverlayCNI() {
				GinkgoWriter.Println("This environment is in underlay mode.")
				Skip("Not applicable to underlay mode")
			}

			if !common.CheckCalicoFeatureOn() {
				GinkgoWriter.Println("The CNI isn't calico.")
				Skip("This case only run in calico")
			}

			cm, err = frame.GetConfigmap("kubeadm-config", "kube-system")
			Expect(err).NotTo(HaveOccurred())

			spc, err = GetSpiderCoordinator(common.SpidercoodinatorDefaultName)
			Expect(err).NotTo(HaveOccurred(), "failed to get SpiderCoordinator, error is %v", err)

			// Switch podCIDRType to `cluster`.
			spcCopy := spc.DeepCopy()
			spcCopy.Spec.PodCIDRType = pointer.String(common.PodCIDRTypeCluster)
			Expect(PatchSpiderCoordinator(spcCopy, spc)).NotTo(HaveOccurred())

			DeferCleanup(func() {
				spc, err := GetSpiderCoordinator(common.SpidercoodinatorDefaultName)
				Expect(err).NotTo(HaveOccurred(), "failed to get SpiderCoordinator, error is %v", err)
				GinkgoWriter.Printf("Display the default spider coordinator information: %+v \n", spc)

				// Switch podCIDRType to `auto`.
				spcCopy := spc.DeepCopy()
				spcCopy.Spec.PodCIDRType = pointer.String(common.PodCIDRTypeAuto)
				Expect(PatchSpiderCoordinator(spcCopy, spc)).NotTo(HaveOccurred())

				Eventually(func() bool {
					spc, err := GetSpiderCoordinator(common.SpidercoodinatorDefaultName)
					Expect(err).NotTo(HaveOccurred(), "failed to get SpiderCoordinator, error is %v", err)
					GinkgoWriter.Printf("Display the default spider coordinator information: %+v \n", spc)

					if spc.Status.OverlayPodCIDR == nil || spc.Status.Phase != coordinatormanager.Synced {
						GinkgoWriter.Printf("status.overlayPodCIDR status is still synchronizing, status %+v \n", spc.Status.OverlayPodCIDR)
						return false
					}

					for _, cidr := range spc.Status.OverlayPodCIDR {
						if ip.IsIPv4CIDR(cidr) {
							if cidr != v4PodCIDRString {
								return false
							}
							GinkgoWriter.Printf("ipv4 podCIDR is as expected, value %v=%v \n", cidr, v4PodCIDRString)
						} else {
							if cidr != v6PodCIDRString {
								return false
							}
							GinkgoWriter.Printf("ipv6 podCIDR is as expected, value %v=%v \n", cidr, v6PodCIDRString)
						}
					}
					return true
				}, common.ExecCommandTimeout, common.ForcedWaitingTime).Should(BeTrue())
			})
		})
```

Reply:
怎么自动cherry-pick 成这样了..